### PR TITLE
powerline v2.3(add new formula)

### PR DESCRIPTION
**Pull request by hyuni**

### CODE REVIEW

File: Library/Formula/powerline.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Documentation: https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/Formula-Cookbook.md
```
Comment (dunn) [inline Library/Formula/powerline.rb]: Remove the boilerplate comments, please!

File: Library/Formula/powerline.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+#                http://www.rubydoc.info/github/Homebrew/homebrew/master/Formula
+# PLEASE REMOVE ALL GENERATED COMMENTS BEFORE SUBMITTING YOUR PULL REQUEST!
+
+class Powerline < Formula
+  desc "Powerline is a statusline plugin for vim, and provides statuslines and prompts for several other applications, including zsh, bash, tmux, IPython, Awesome and Qtile."
```
Comment (dunn) [inline Library/Formula/powerline.rb]: This'll need to be much shorter; 80 characters including the formula name.

File: Library/Formula/powerline.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  homepage "https://github.com/powerline/powerline"
+  url "https://github.com/powerline/powerline/archive/2.3.tar.gz"
+  version "2.3"
```
Comment (dunn) [inline Library/Formula/powerline.rb]: Homebrew will be able to detect the version here from the URL string, so this isn't necessary.  Running `brew audit --strict --online powerline` will help you catch these things.

File: Library/Formula/powerline.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  sha256 "c65762df4733afed585af722978d24bc555fbbabb8ae707e4e525dff6412de5f"
+
+  depends_on :python if MacOS.version <= :snow_leopard
+
+  def install
+      system "python", *Language::Python.setup_install_args(prefix)
+
+      ENV.prepend_create_path "PYTHONPATH", libexec/"lib/python2.7/site-packages"
+      system "python", *Language::Python.setup_install_args(libexec)
```
Comment (dunn) [inline Library/Formula/powerline.rb]: `setup.py` is being run twice, once here and above at line 15.  Is that necessary?

**Comment by hyuni** [comment]
@dunn thanks, update formula and new pull request.

File: Library/Formula/powerline.rb (hunk 4 above)
Comment (hyuni) [inline Library/Formula/powerline.rb]: "python for formula authors.md" has Installing bindings.
is there another way?

File: Library/Formula/powerline.rb [hunk 5]
```diff
@@ -0,0 +1,23 @@
+class Powerline < Formula
+  desc "statusline plugin for vim, and provides statuslines & prompts"
+  homepage "https://github.com/powerline/powerline"
+  url "https://github.com/powerline/powerline/archive/#{version}.tar.gz"
```
Comment (mikemcquaid) [inline Library/Formula/powerline.rb]: Inline the version here.

File: Library/Formula/powerline.rb [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+    ENV.prepend_create_path "PYTHONPATH", libexec/"lib/python2.7/site-packages"
+    system "python", *Language::Python.setup_install_args(libexec)
+
+    bin.install Dir[libexec/"bin/*"]
+    bin.env_script_all_files(libexec/"bin", :PYTHONPATH => ENV["PYTHONPATH"])
+  end
+
+  test do
+    system "#{bin}/powerline", "--help"
```
Comment (mikemcquaid) [inline Library/Formula/powerline.rb]: Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

File: Library/Formula/powerline.rb [hunk 7]
```diff
@@ -0,0 +1,22 @@
+class Powerline < Formula
+  desc "statusline plugin for vim, and provides statuslines & prompts"
+  homepage "https://github.com/powerline/powerline"
+  url "https://github.com/powerline/powerline/archive/2.3.tar.gz"
+  sha256 "c65762df4733afed585af722978d24bc555fbbabb8ae707e4e525dff6412de5f"
+
+  depends_on :python if MacOS.version <= :snow_leopard
+
+  def install
+    system "python", *Language::Python.setup_install_args(prefix)
```
Comment (bfontaine) [inline Library/Formula/powerline.rb]: Is this line necessary given that there’s almost the same with `libexec` below?

Comment (hyuni) [inline Library/Formula/powerline.rb]: test :
remove #10 line and brew install powerline. 
but don't exist /usr/local/lib/python2.7/site-packages/powerline folder.

File: Library/Formula/powerline.rb (hunk 6 above)
Comment (hyuni) [inline Library/Formula/powerline.rb]: @mikemcquaid thanks. read powerline document and searching better test.

File: Library/Formula/powerline.rb (hunk 7 above)
Comment (hyuni) [inline Library/Formula/powerline.rb]: @bfontaine thanks. update this formula.

**Comment by apjanke** [comment]
Merged. Thank you for your contribution to Homebrew!

**Comment by hyuni** [comment]
@apjanke Thanks

**Comment by sbusch** [comment]
AFAIK the Homebrew policy disencourages creation of formulas which can already be installed by another packager (like pip). Although this formula doesn't hurt me in any way, I'm just wondering whether this has changed?

See "Bindings" section of https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/Acceptable-Formulae.md

**Comment by mikemcquaid** [comment]
@sbusch It's an exception as it's an application rather than a library: https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/Acceptable-Formulae.md#we-try-hard-to-avoid-dupes-in-homebrewhomebrew

**Comment by tdsmith** [comment]
Note for posterity: this formula should have followed the Applications section of the documentation instead of installing directly to `prefix`, which generally should not happen in homebrew-core: https://github.com/Homebrew/brew/blob/master/share/doc/homebrew/Python-for-Formula-Authors.md#applications; ref Homebrew/homebrew-core#1767

**Comment by tdsmith** [comment]
Boneyarded in https://github.com/Homebrew/homebrew-core/pull/1804 because powerline installed anywhere else just isn't very useful since it needs to be importable for nearly any of the functionality to work except bash integration.

**Pull request closed without merging**